test/test_releasenotes.py

```diff
@@ -0,0 +1,113 @@
+        name = self.release.component_name
+        self.assertEqual(name, "Stream Engine")
+
+    def test_get_latest_version(self):
+        version_number = self.release.latest_version
+        self.assertEqual(version_number, "1.10.0")
+
+    def test_get_latest_date(self):
+        date = self.release.latest_date
+        self.assertEqual(date, "2019-07-19")
+
+    def test_get_latest_descriptor(self):
+        descriptor = self.release.latest_descriptor
+        self.assertEqual(descriptor, "Development Release")
+
+    def test_get_component(self):
+        component = self.release.component
+        self.assertIsNotNone(component)
+
+    def test_get_dependencies(self):
+        dependencies = self.release.dependencies
+        package_names = [dependency.split("=")[0] for dependency in dependencies]
+        self.assertIsNotNone(dependencies)
+        self.assertIn("mock", package_names)
+
+    def test_component_serialization(self):
+        component = self.release.component
+        serialized_component = json.dumps(component, cls=ComponentDecoder, indent=2, separators=(',', ': '))
+        self.assertIsNotNone(serialized_component)
+
+    def test_known_version(self):
+        component = self.release.component
+        version = component.versions[4]
+        self.assertEqual(version.descriptor, "Release")
+        self.assertEqual(version.number, "1.6.0")
+        self.assertEqual(version.date, "2018-07-20")
+        self.assertIsNone(version.details)
+        self.assertIs(len(version.notes), 5)
+
+    def test_known_note(self):
+        component = self.release.component
+        version = component.versions[4]
+        note = version.notes[0]
+        self.assertEqual(note.title, "Issue #13448 - Prevent getting fill values for wavelength coordinate")
+        self.assertEqual(note.ticket, "13448")
+        self.assertEqual(note.details, "- Correct code error that led to NaNs for wavelength values\n- Add handling "
+                                       "to replace coordinate fill values with whole number sequence")
+
+    def test_version_without_date(self):
+        component = self.release.component
+        version = component.versions[18]
+        self.assertEqual(version.number, "1.3.0")
+        self.assertEqual(version.date, "")
+
+    def test_version_without_notes(self):
+        component = self.release.component
+        version = component.versions[18]
+        self.assertEqual(version.number, "1.3.0")
+        self.assertFalse(version.notes)
+        self.assertEqual(version.details, "Updated required packages\n- numpy\n- numexpr\n- scipy\n- ooi-data\n- "
+                                          "flask\n- gunicorn\n- xarray\n- pandas")
+
+    def test_note_without_details(self):
+        component = self.release.component
+        version = component.versions[20]
+        note = version.notes[0]
+        self.assertEqual(version.number, "1.2.9")
+        self.assertIsNone(note.details)
```

Comment:
wouldn't the interface be cleaner returning an empty list `''` instead of None for an expected string?

Reply:
Replacing use of None for details default value with an empty String for default value. Updated test cases and verified this works with the uFrame interface.